tests/js/snapshots.test.js

```diff
@@ -1,9 +1,168 @@
+
+test.describe("Qiskit top nav bar", () => {
+  test("does not cover Furo's top nav bar when scrolled down", async ({
+    page,
+  }) => {
+    await page.goto("sphinx_guide/lists.html");
+    await scrollDown(page, 200);
+
+    const pageToCVisible = await isVisibleInViewport(
+      page,
+      "div.toc-title-container"
+    );
+    expect(pageToCVisible).toBe(true);
+
+    const searchVisible = await isVisibleInViewport(
+      page,
+      "input.sidebar-search"
+    );
+    expect(searchVisible).toBe(true);
+
+    await setMobile(page);
+    const mobileHeaderVisible = await isVisibleInViewport(
+      page,
+      "header.mobile-header"
+    );
+    expect(mobileHeaderVisible).toBe(true);
+  });
+
+  test("does not cover the top of # anchor links", async ({ page }) => {
+    const checkHeader = async () => {
+      await page.goto("sphinx_guide/lists.html#definition-lists");
+      // We have to wait for the page animation to update.
+      await page.waitForTimeout(800);
+      const headerVisible = await isVisibleInViewport(
+        page,
+        "section#definition-lists > h2"
+      );
+      expect(headerVisible).toBe(true);
+    };
+
+    await checkHeader();
+
+    await setMobile(page);
+    // Go to a new page first to reset the scroll state.
+    await page.goto("");
+    await checkHeader();
+  });
+
+  test("does not cover the side menus when expanded on mobile", async ({
+    page,
+  }) => {
+    await setMobile(page);
+    await page.goto("");
+
+    const click = async (selector) => {
+      await page.locator(selector).click();
+      // Wait for the page to update.
+      await page.waitForTimeout(500);
+    };
+
+    await click("div.header-right label.toc-overlay-icon i");
+    const pageToCVisible = await isVisibleInViewport(
+      page,
+      "div.toc-title-container"
+    );
+    expect(pageToCVisible).toBe(true);
+
+    // Reload the page to close the nav bar.
+    await page.goto("");
+
+    await click("div.header-left i");
+    const searchVisible = await isVisibleInViewport(
+      page,
+      "input.sidebar-search"
+    );
+    expect(searchVisible).toBe(true);
+  });
+});
+
+test.describe("Furo top nav bar", () => {
+  test("uses custom page ToC icon on tablet", async ({ page }) => {
+    await setTablet(page);
+    await page.goto("");
+    const pageToC = page.locator("div.content-icon-container");
+    await expect(pageToC).toHaveScreenshot();
+  });
+
+  test("uses custom icon and text on mobile", async ({ page }) => {
+    await setMobile(page);
+    await page.goto("");
+    const header = page.locator("header.mobile-header");
+    await expect(header).toHaveScreenshot();
+  });
+});
+
 test.describe("footer", () => {
   test("includes page analytics", async ({ page }) => {
     await page.goto("");
     const footer = page.locator("footer");
     await expect(footer).toHaveScreenshot();
   });
+
+  test("shows analytics options in purple when hovered", async ({ page }) => {
```

Comment:
for this test and the "thank-you" test I am wondering if we could combine this into one test and reduce the number of screenshots we need. For example to test the purple colour we can check the element has a specific css class with a specific colour. Then for the thank-you we can check first that the element is not visible then that the element has become visible. 

I think having one screenshot for this component is good, but I don't think we need one for each of these specific features. Also my understanding of snapshot tests (correct me if I'm wrong) is that they are designed to test the overall UI of a component, rather than functionality (which can normally be tested with other means)

Reply:
I think that here the screenshots could be useful because we do want to ensure the _styling_ is correct for these two edge cases. That we are positioning things correctly, have the right font, etc. It took a few specific CSS rules to handle those edge cases:

https://github.com/Qiskit/qiskit_sphinx_theme/blob/f5b9bbc7c91714cce8060ab5aa06fba5dd9dbc84/qiskit_sphinx_theme/furo/base/static/styles/qiskit_changes.css#L77-L84

https://github.com/Qiskit/qiskit_sphinx_theme/blob/f5b9bbc7c91714cce8060ab5aa06fba5dd9dbc84/qiskit_sphinx_theme/furo/base/static/styles/qiskit_changes.css#L93-L97

Basically, how I decided what to test was anywhere we made a change to upstream Furo. These two edge cases match that. And they're good examples of why I like tests -- I'd never normally remember to manually test them both because they're so niche.

Reply:
I still think essentially having 3 screenshots that capture the helpful component is overkill tbh. We really only need to test 2 lines of CSS to sufficiently test the edge cases you are describing, and I think wherever possible we should be testing things in code and use snapshots minimally. We can see that the majority of the styling is ok from the "footer includes page analytics" screenshot. If you are worried about the "thank-you" displaying correctly then maybe keep the screenshot for that, but I think the hover one could definitely be removed